lib/dialects/mysql/query.js

```diff
@@ -110,6 +110,8 @@ Query.prototype.formatResults = function(data) {
     result = data[0].version;
   } else if (this.isForeignKeysQuery()) {
     result = data;
+  } else if (this.isInsertQuery() || this.isUpdateQuery()) {
+    result = [result, data.affectedRows];
```

Comment:
You do need to affect the `isInsertQuery()` because the `Instance.save()` uses Insert/Update interchangeably depending on if it's a new instance.